src/analysis/cauchy_equation.lean

```diff
@@ -0,0 +1,305 @@
+/-
+Copyright (c) 2022 Mantas Bakšys. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mantas Bakšys
+-/
+import analysis.normed_space.pointwise
+import measure_theory.measure.haar_lebesgue
+
+/-!
+# Cauchy's Functional Equation
+
+This file contains the classical results about the Cauchy's functional equation
+`f (x + y) = f x + f y` for functions `f : ℝ → ℝ`. In this file, we prove that the solutions to this
+equation are linear up to the case when `f` is a Lebesgue measurable functions, while also deducing
+intermediate well-known variants.
+-/
+
+open add_monoid_hom measure_theory measure_theory.measure metric nnreal set
+open_locale pointwise topological_space
+
+variables {ι : Type*} [fintype ι]
+
+local notation `ℝⁿ` := ι → ℝ
+
+theorem cauchy_rational_add (f : ℝ →+ ℝ) :
```

Comment:
```suggestion
/-- **Cauchy's functional equation**. An additive monoid homomorphism automatically preserves `ℚ`.
-/
theorem add_monoid_hom.is_linear_map_rat (f : ℝ →+ ℝ) :
```
Can you also put it closer to [`add_monoid_hom.map_rat_cast_smul`](https://leanprover-community.github.io/mathlib_docs/algebra/module/basic.html#add_monoid_hom.map_rat_cast_smul) and generalize `ℝ` accordingly?

Reply:
I can but I just noticed that [add_monoid_hom.to_rat_linear_map](https://leanprover-community.github.io/mathlib_docs/algebra/module/linear_map.html#add_monoid_hom.to_rat_linear_map) already exists. I am now confused if we now even want the `is_linear_map is_linear_map ℚ f` version. @sgouezel, perhaps you have any thoughts on this?